src/mrpro/algorithms/_adam.py

```diff
@@ -20,7 +22,7 @@
 
 def adam(
     f: Operator[*tuple[torch.Tensor, ...], tuple[torch.Tensor]],
```

Comment:
We used to mix tuple and int for parameters. This causes issues if we call it with the other one.
So 
```
def f(a:tuple[int,...])-> None:
    ...
f([1])
```
would make mypy complain. 

Sequence is anything that has getitem and has a length, so list and tuple both work.
So we should always use Sequence for inputs. This also means we cannot modify it (as it doesnt specify that it has a setitem). So its a tuple or a list treated like a tuple -> good for inputs. Otherwise the caller has to convert to a tuple.


Our return type is more narrow, we actually write what we return. We return tuples for operators.

The `*tuple[torch.Tensor, ...]` means that it is a function with more than one input. Nowhere in the inputs is a tuple involved, it is just a way to write `Operator: tensor,tensor,tensor,... -> (tensor,)`

This is only required if we want to write "arbittrary many inputs", otherwise it is, for example:
`Operator[torch.Tensor, tuple[torch.Tensor]], Operator[torch.Tensor, torch.Tensor, tuple[torch.Tensor]],`
This is the same style of generic typing as used for `Callable`.

So, long story short:
Sequence for inputs.
unpacked tuple in the typhints for operator (and callable) if we want to use an ellipsis.
tuple as the return type.

Reply:
> tuple as the return type.

Why does adam then return a list?

Reply:
good point.

Reply:
fixed return type(s) for adam and lbfgs